src/main/resources/swagger/api-docs.yaml

```diff
@@ -371,5 +379,49 @@ definitions:
           Optional bucket URI to an archive containing Jupyter notebook extension files.
           The archive must be in tar.gz format, must not include a parent directory, and
           must have an entry point named 'main'. For more information on notebook extensions,
-          see: http://jupyter-notebook.readthedocs.io/en/latest/extending/frontend_extensions.html.
+          see http://jupyter-notebook.readthedocs.io/en/latest/extending/frontend_extensions.html.
+      machineConfig:
+        description: The configurations for the machine and worker nodes
+        $ref: '#/definitions/MachineConfig'
 
+  MachineConfig:
+    description: ''
+    properties:
+      numberOfWorkers:
+        type: integer
+        description: |
+          Optional, number of workers in the cluster. For a SingleNode, this field should not be defined, as the
+          SingleNode clusters have 0 workers by default. For a Standard cluster, workers should be 2 or more.
+      masterMachineType:
+        type: string
+        description: |
+          Optional, the machine type determines the number of CPUs and memory for the master node. For example "n1-standard-16"
+          or "n1-highmem-64". If unspecified, defaults to creating a "n1-standard-4" machine. To decide which is right for you,
+          see https://cloud.google.com/compute/docs/machine-types
+      masterDiskSize:
+        type: integer
+        description: |
+          Optional, the size of the disk on the master node. Minimum size is 10GB.
```

Comment:
Our minimum is 100 GB.  Hail requires 50 GB and we specify 100 GB in `spark-defaults.conf`.

Reply:
Should also validate this?

Reply:
We can change the minimum to 100GB and throw an exception if it's less

Reply:
if they write something less, just fix it (and document that anything <100GB will be interpreted as 100GB)

Reply:
That's just for the master, not the worker, right?

Reply:
I believe it's both

Reply:
So, for any value less than 100, we want 100, but if it's None, we set it to 500 (current default)? 

Can we just return the default (500) if it's less than 100, or just change the default to 100?

Reply:
That first sentence sounds right. If you don't say anything, we give you the default. If you give us a number lower than the minimum, we make it the minimum.

Reply:
I leave questions about what the default _should_ be to more knowledgeable people :)